Trappist/src/Promact.Trappist.Web/wwwroot/app/questions/questions-programming/questions-programming.component.ts

```diff
@@ -1,9 +1,168 @@
+import { DifficultyLevel } from '../enum-difficultylevel';
+import { MdSnackBar } from '@angular/material';
+import { Router } from '@angular/router';
 
 @Component({
     moduleId: module.id,
     selector: 'questions-programming',
     templateUrl: 'questions-programming.html'
 })
 
-export class QuestionsProgrammingComponent { }
+export class QuestionsProgrammingComponent implements OnInit {
+
+    selectedLanguageList: string[];
+    codingLanguageList: string[];
+    categoryList: Category[];
+    questionModel: QuestionModel;
+
+    nolanguageSelected: boolean;
+    showTestCase: boolean;
+    collapseTestCase: boolean;
+    collapseNewTestCase: boolean;
+    showNewTestCase: boolean;
+    isCategoryReady: boolean;
+    isLanguageReady: boolean;
+    isFormSubmitted: boolean;
+    code: any;
+
+    private successMessage: string = 'Question saved successfully';
+    private failedMessage: string = 'Question failed to save';
+    private routeToDashboard = '/questions';
+
+    constructor(private questionsService: QuestionsService,
+        private categoryService: CategoryService,
+        private snackBar: MdSnackBar,
+        private router: Router) {
+
+        this.nolanguageSelected = true;
+        this.isCategoryReady = false;
+        this.isLanguageReady = false;
+        this.selectedLanguageList = new Array<string>();
+        this.codingLanguageList = new Array<string>();
+        this.categoryList = new Array<Category>();
+        this.questionModel = new QuestionModel();
+    }
+
+    ngOnInit() {
+        this.getCodingLanguage();
+        this.getCategory();
+    }
+
+    /**
+     * Gets all the coding languages
+     */
+    private getCodingLanguage() {
+        this.questionsService.getCodingLanguage().subscribe((response) => {
+            this.codingLanguageList = response;
+            this.codingLanguageList.sort();
+            this.isLanguageReady = true;
+        });
+    }
+
+    /**
+     * Gets all the categories
+     */
+    private getCategory() {
+        this.categoryService.getAllCategories().subscribe((response) => {
+            this.categoryList = response;
+            this.isCategoryReady = true;
+        });
+    }
+
+    /**
+     * Adds language to the selectedLanguageList
+     * @param language : language to add
+     */
+    selectLanguage(language: string) {
+        let index = this.selectedLanguageList.indexOf(language);
+        if (index === -1) {
+            this.selectedLanguageList.push(language);
+            this.codingLanguageList.splice(this.codingLanguageList.indexOf(language), 1);
+        }
+        this.checkIfNoLanguageSelected();
+    }
+
+    /**
+     * Removes language from selectedLanguageList 
+     * @param language : Language to remove
+     */
+    removeLanguage(language: string) {
+        this.selectedLanguageList.splice(this.selectedLanguageList.indexOf(language), 1);
+        this.codingLanguageList.push(language);
+        this.checkIfNoLanguageSelected();
```

Comment:
The languages are stored in
`codingLanguageList: string[];`
When the user selects a language it get pushed in 
`selectedLanguageList: string[];`
The md-chip is populated using this array. There is an option in md-chip to remove the current chip. When user removes the chip this function is called which removes it from selectedLanguageList and add back to the codingLanguageList.